packages/core/src/auth/secondaryAuth.ts

```diff
@@ -147,6 +147,40 @@ export class SecondaryAuth<T extends Connection = Connection> {
                 await this.clearSavedConnection()
             }
         })
+
+        const refreshConn = (event: string) => {
+            getLogger().debug(`secondaryAuth: detected ${event} event in sso cache, refreshing auth.`)
+            globals.clock.setTimeout(
+                telemetry.function_call.run(
+                    () => async () => {
+                        if (this.#savedConnection?.id === this.getStateConnectionId()) {
+                            // Someone updated our cache but the global state doesn't indicate anything new, so do nothing.
+                            // (it could have been us that updated the cache)
+                            getLogger().debug(
+                                `secondaryAuth: cache event did not update global state, no refresh is needed.`
+                            )
+                            return
+                        }
+                        await this.auth.restorePreviousSession()
+                        await this.restoreConnection(true)
```

Comment:
What is the reason for restorePreviousSession as well as restoreConnection? restoreConnection eventually calls restorePreviousSession. It goes restoreConnection -> tryAutoConnect -> restorePreviousSession. And tryAutoConnect uses once(), maybe this was causing things to not work?

Reply:
`tryAutoConnect ` is a once() function that will likely have been called on startup. So it will be skipped by `restoreConnection` when this fires. To avoid adding the `force = true` pattern (and other side effects) to that function as well, we can just call the important part of that function manually.

Reply:
I would like to avoid putting a comment describing that because I think it will add confusion. This question is rather niche to our understanding of how the auth system works. Let me know if you think otherwise